### DISCUSSION TITLE
How to convert yolox pth to ncnn ?

### DISCUSSION THREAD

**Discussion by danurrahmajati** [Megvii-BaseDetection/YOLOX#1124] [Q&A]
i try to convert using onnx but generate lots of error, can you show me how to convert to ncnn ?

**Comment by FateScript** [comment]
Could you tell me which step stops you from this [tutorial](https://github.com/Megvii-BaseDetection/YOLOX/tree/main/demo/ncnn/cpp)?

**Reply by danurrahmajati** [reply]
Thanks for the quick replay, there is more error than expected in step 3. I think that is because I suppose to train data for yolox s, instead i use another normal weight model.

**Reply by FateScript** [reply]
Does `build/tools/ncnn/onnx2ncnn` in ncnn generate bin file ? Any log provided?

**Reply by danurrahmajati** [reply]
step 2
``cd <path of yolox>
python3 tools/export_onnx.py -n yolox-s``

did i miss something, I don't know where I suppose to put my training model, there is no parameter to put pth file.

**Reply by FateScript** [reply]
did your download the network weight? What's your error log?

**Reply by danurrahmajati** [reply]
I train my own custom data set using yolox.nano model and convert it to ncnn model. I hope i can run it on android device

**Reply by FateScript** [reply]
Ok, could you plz provide your error log?

**Reply by danurrahmajati** [reply]
ONNX conversion is working well but onnx2ncnn return this error

```
D:\ncnn\build\tools\onnx> ./onnx2ncnn yolo_nano.onnx model.param model.bin
open failed yolo_nano.onnx
read_proto_from_binary failed
```

the example  like this 
```
cd build/tools/ncnn
./onnx2ncnn yolox.onnx model.param model.bin
```

but i cannot find tools/ncnn folder .... the only reasonable folder is onnx

**Reply by danurrahmajati** [reply]
Sorry my bad, set the wrong file name

**Reply by danurrahmajati** [reply]
i think my export failed since the detection not working. Here are my file [https://drive.google.com/file/d/17yLMbI_isTPyXMNs1BtZSnxm5U2YDJ4x/view?usp=sharing, https://drive.google.com/file/d/1MhOOCUXgx5qborRkingBSDgKSPbEwht0/view?usp=sharing](url) 

its about road damage detector, 
i have 4 classes

D00
D10
D20
D40
I don't know the problem yet
